### PR TITLE
Remove unnecessary items from the PR checklist

### PR DESCRIPTION
**Pull request by bessbd**
The checklist includes items about writing documentation and running automated tests.

The first item is not applicable as most of the commits in this repo are supposed to be doc changes and the automated tests are run for every PR via the Jenkins CI so it's redundant to have a checkbox in the PR summary.

This change removes those two non-applicable items.

## Checklist

- [ ] Documentation is written and is accurate;
- [ ] `make check` passes with no errors
- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/master/rebar.config.script) with the commit hash once this PR is rebased and merged

### CODE REVIEW

**Comment by janl** [comment]
`make check` passes is useful to ensure that you don’t accidentally commit a syntax error. You would likely want a faster turnaround time on those than a CI build. So we recommend you run `make check` locally before opening a PR. Hence the checkbox.

**Comment by bessbd** [comment]
> `make check` passes is useful to ensure that you don’t accidentally commit a syntax error. You would likely want a faster turnaround time on those than a CI build. So we recommend you run `make check` locally before opening a PR. Hence the checkbox.

What do you think about adding a md-comment notice then? Eg. 
```
<!-- Before opening the PR, consider running `make check` locally for a faster turnaround time -->
```
?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -36,6 +36,4 @@
 
 ## Checklist
 
-- [ ] Documentation is written and is accurate;
-- [ ] `make check` passes with no errors
```
Comment (nickva) [inline .github/PULL_REQUEST_TEMPLATE.md:40]: Let's keep `make check` it runs the linter to check formatting and such

**Comment by janl** [comment]
sounds good to me!

**Comment by bessbd** [comment]
> sounds good to me!

I've added the comment, @janl . Let me know what you think.

**Comment by wohali** [comment]
@bessbd Thank you for your first contribution (that I'm aware of!) to Apache CouchDB! :trumpet:

**Comment by bessbd** [comment]
> Personally I prefer it if folks can run `make check` locally, but I understand that it'll be easier for drive-by doc improvements if this isn't a mandatory part of the process.
> 
> +1

I think it's useful that people execute `make check` locally. I just didn't think manually ticking the checkmark adds a lot of value.

**Comment by bessbd** [comment]
> @bessbd Thank you for your first contribution (that I'm aware of!) to Apache CouchDB! 🎺

Thank you @wohali ! (And thanks for the approval and merge, too!) I've made a few changes - even in this repo: https://github.com/apache/couchdb-documentation/commits?author=bessbd , but the kind words feel good nevertheless!

**Pull request merged**